weekDownloader.py

```diff
@@ -43,6 +43,41 @@ def get_only_video_steps(step_list, token):
     return resp_list
 
 
+def batch_main(argv):
+    if len(argv) != 4:
+        print('Input Error, pass 4 arguments. first - Course_id, '
+              'second - Week index, third - folder name, forth - quality.')
+        sys.exit('1')
+
+    """
+    Example how to receive token from Stepic.org
+    Token should also been add to every request header
+    example: requests.get(api_url, headers={'Authorization': 'Bearer '+ token})
+    """
+
+    auth = HTTPBasicAuth(client_id, client_secret)
+    resp = requests.post('https://stepic.org/oauth2/token/', data={'grant_type': 'client_credentials'}, auth=auth)
+    token = json.loads(resp.text)['access_token']
+
+    course_data = get_course_page('http://stepic.org/api/courses/' + argv[0], token)
+    all_units = get_unit_list(get_all_weeks(course_data), token)
+    all_steps = get_steps_list(all_units, int(argv[1]), token)
+    only_video_steps = get_only_video_steps(all_steps, token)    
+    url_list_with_q = [url for x in only_video_steps for url in x['video']['urls'] if url['quality'] == argv[3]]
```

Comment:
> Also I think in case the is no selected video, you could get a warning message in the console and download the best availible quality as it was previously.
Will it be OK?

Sounds good.

This script is used by me to create concated videos for weeks in courses and than upload them to Stepic. My colleagues also using it, so, what they wouldnt like -- is to edit ENV variables and source code for downloading every course. I think best idea to run this script will be passing args from command line.